contracts/contracts/layer_0/clearing/ClearingStorageWrapper2.sol

```diff
@@ -334,22 +335,22 @@ abstract contract ClearingStorageWrapper2 is
             _protectedClearingOperation.clearingOperation,
             _amount,
             _protectedClearingOperation.from,
-            _msgSender(),
-            false,
-            ''
+            '',
+            address(0),
+            ThirdPartyType.PROTECTED
         );
     }
 
     function _operateClearingCreation(
         IClearing.ClearingOperation memory _clearingOperation,
         address _from,
-        address _operator,
-        bool _reduceAllowance,
+        address _thirdPartyAddress,
+        ThirdPartyType _thirdPartyType,
         uint256 _amount,
         IClearing.ClearingOperationType _operationType
     ) internal returns (uint256 clearingId_) {
-        if (_reduceAllowance) {
-            _decreaseAllowedBalance(_from, _operator, _amount);
+        if (_thirdPartyType == ThirdPartyType.AUTHORIZED) {
+            _decreaseAllowedBalance(_from, _thirdPartyAddress, _amount);
```

Comment:
The allowance normally is decreased to the msg.sender. 

Check transferFrom logic of ERC20 to review it, pls.